### PR TITLE
[BD-46] docs: improve documentation for various components

### PR DESCRIPTION
**Pull request by viktorrusakov**
Improve documentation for Badge, Breadcrumbs, Button, Button Group, Card, Carousel, Collapsible, Container, Fullscreen modal, icon components and Form subcomponents.

JIRA: [PAR-578](https://openedx.atlassian.net/browse/PAR-578)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @viktorrusakov! I've created [BLENDED-1021](https://openedx.atlassian.net/browse/BLENDED-1021) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: src/Button/button-group.mdx [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 title: 'Button Group'
 type: 'component'
```
Comment (adamstankiewicz) [inline src/Button/button-group.mdx:3]: [clarification] Is this button-group.mdx documentation actually reachable by the technical documentation site navigation? There's obviously a page for the `Button` but I'm not seeing this page for ButtonGroup/ButtonToolbar in the navigation. Do you think it should be?

File: src/Collapsible/CollapsibleAdvanced.jsx [hunk 2]
```diff
@@ -93,10 +93,15 @@ class CollapsibleAdvanced extends React.Component {
 CollapsibleAdvanced.propTypes = {
   children: PropTypes.node,
```
Comment (adamstankiewicz) [inline src/Collapsible/CollapsibleAdvanced.jsx:94]: Should there also be a comment explaining `children` and `className` props?

File: src/Collapsible/CollapsibleBody.jsx [hunk 3]
```diff
@@ -25,7 +25,9 @@ function CollapsibleBody({
 
 CollapsibleBody.propTypes = {
   children: PropTypes.node,
```
Comment (adamstankiewicz) [inline src/Collapsible/CollapsibleBody.jsx:27]: Should there be a comment for `children`?

File: src/Collapsible/CollapsibleTrigger.jsx [hunk 4]
```diff
@@ -52,10 +52,15 @@ function CollapsibleTrigger({
 
 CollapsibleTrigger.propTypes = {
   children: PropTypes.node,
```
Comment (adamstankiewicz) [inline src/Collapsible/CollapsibleTrigger.jsx:54]: Should there be a comment for `children`?

File: src/Collapsible/CollapsibleVisible.jsx [hunk 5]
```diff
@@ -19,7 +19,9 @@ function CollapsibleVisible({
 
 CollapsibleVisible.propTypes = {
   children: PropTypes.node,
```
Comment (adamstankiewicz) [inline src/Collapsible/CollapsibleVisible.jsx:21]: Should there be a comment for `children`?

File: src/Collapsible/index.jsx [hunk 6]
```diff
@@ -53,14 +53,23 @@ const Collapsible = React.forwardRef((props, ref) => {
 Collapsible.propTypes = {
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Collapsible/index.jsx:55]: Should there be a comment for `children` and `className`?

File: src/Form/FormCheckbox.jsx [hunk 7]
```diff
@@ -33,6 +33,7 @@ const CheckboxControl = React.forwardRef(
 );
 
 CheckboxControl.propTypes = {
+  /** Specifies whether the checkbox should be rendered in indeterminate state. */
   isIndeterminate: PropTypes.bool,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormCheckbox.jsx:38]: Should there be a comment for `className`?

File: src/Form/FormCheckbox.jsx [hunk 8]
```diff
@@ -99,11 +100,17 @@ const FormCheckbox = React.forwardRef(({
 FormCheckbox.propTypes = {
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormCheckbox.jsx:102]: Should there be a comment for `children` and `className`?

File: src/Form/FormCheckboxSet.jsx [hunk 9]
```diff
@@ -37,12 +37,19 @@ const FormCheckboxSet = ({
 FormCheckboxSet.propTypes = {
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormCheckboxSet.jsx:39]: Should there be a comment for `children` and `className`?

File: src/Form/FormControl.jsx [hunk 10]
```diff
@@ -60,25 +59,36 @@ const FormControl = React.forwardRef(({
   );
 });
 
+const SIZE_CHOICES = ['sm', 'lg'];
+
 FormControl.Feedback = FormControlFeedback;
 FormControl.Description = FormControlFeedback;
 
 FormControl.propTypes = {
+  className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormControl.jsx:68]: Should there be a comment for `className`?

File: src/Form/FormControlFeedback.jsx [hunk 11]
```diff
@@ -21,12 +20,26 @@ const FormControlFeedback = ({ children, ...props }) => {
...
+  'criteria-valid',
+  'criteria-invalid',
+];
+
 FormControlFeedback.propTypes = {
-  hasIcon: PropTypes.bool,
-  type: PropTypes.oneOf(Object.values(FORM_TEXT_TYPES)),
-  icon: PropTypes.node,
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormControlFeedback.jsx:35]: Should there be a comment for `children` and `className`?

File: src/Form/FormGroup.jsx [hunk 12]
```diff
@@ -26,17 +25,22 @@ const FormGroup = ({
   </FormGroupContextProvider>
 ));
 
+const SIZE_CHOICES = ['sm', 'lg'];
+
 FormGroup.propTypes = {
-  as: PropTypes.elementType,
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormGroup.jsx:32]: Should there be a comment for `children` and `className`?

File: src/Form/FormLabel.jsx [hunk 13]
```diff
@@ -20,14 +20,15 @@ const FormLabel = ({ children, isInline, ...props }) => {
...
+const SIZE_CHOICES = ['sm', 'lg'];
+
 FormLabel.propTypes = {
-  isInline: PropTypes.bool,
-  size: PropTypes.oneOf([
-    FORM_CONTROL_SIZES.SMALL,
-    FORM_CONTROL_SIZES.LARGE,
-  ]),
   className: PropTypes.string,
   children: PropTypes.node.isRequired,
```
Comment (adamstankiewicz) [inline src/Form/FormLabel.jsx:27]: Should there be a comment for `children` and `className`?

File: src/Form/FormRadio.jsx [hunk 14]
```diff
@@ -72,10 +72,15 @@ const FormRadio = React.forwardRef(({
 FormRadio.propTypes = {
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormRadio.jsx:74]: Should there be a comment for `children` and `className`?

File: src/Form/FormSwitch.jsx [hunk 15]
```diff
@@ -35,6 +35,7 @@ const SwitchControl = React.forwardRef(
 );
 
 SwitchControl.propTypes = {
+  /** Specifies whether input should be rendered in indeterminate state. */
   isIndeterminate: PropTypes.bool,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormSwitch.jsx:40]: Should there be a comment for `className`?

File: src/Form/FormText.jsx [hunk 16]
```diff
@@ -71,18 +71,32 @@ const FormText = ({
...
+  'criteria-valid',
+  'criteria-invalid',
+];
+
 FormText.propTypes = {
-  hasIcon: PropTypes.bool,
-  type: PropTypes.oneOf(Object.values(FORM_TEXT_TYPES)),
-  icon: PropTypes.node,
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Form/FormText.jsx:86]: Should there be a comment for `children` and `className`?

File: src/Icon/README.md [hunk 17]
```diff
@@ -11,7 +11,7 @@ devStatus: 'TBD'
 notes: 'Reconcile purpose of this component with use of FontAwesome'
 ---
 
-Displays an svg icon from `@edx/paragon/icons`. See  Icons Foundation Documentation for a list of all available icons.
+Displays an svg icon from `@edx/paragon/icons`. See [Icons Foundation Documentation](/foundations/icons) for a list of all available icons.
```
Comment (adamstankiewicz) [inline src/Icon/README.md:14]: 🎉

File: src/Modal/FullscreenModal.jsx [hunk 18]
```diff
@@ -22,18 +22,34 @@ const FullscreenModal = ({
 
 FullscreenModal.propTypes = {
   children: PropTypes.node.isRequired,
+  className: PropTypes.string,
```
Comment (adamstankiewicz) [inline src/Modal/FullscreenModal.jsx:25]: Should there be a comment for `children` and `className`?

**Review by adamstankiewicz — COMMENTED** [review]
Looks good. Just a question of whether there should be comments describing `children` and `className` props throughout.

I also had a clarifying question about `button-group.mdx`. Am I right in thinking this page is not actually discoverable from the technical documentation navigation?

File: src/Button/button-group.mdx (hunk 1 above)
Comment (viktorrusakov) [reply src/Button/button-group.mdx:3]: You could reach it in `All components (A-Z)` section at the bottom of the documentation (that's how I found it actually, I was going in alphabetical order to update the components), but it didn't belong to any other category which is weird. I've added it to `Buttonlike` category for now, I believe it definitely belongs there.

I'm wondering if we need a separate page for ButtonToolbar component, seems like it's closely connected to the ButtonGroup and one page should be enough. Do you think we should add another page for it?

**Comment by viktorrusakov** [comment]
I've updated all the missing props descriptions for `className` and `children`. Not sure how it happened, I think I saw that one of the components had descriptions for all props except these ones and I thought maybe they are self-explanatory and I could skip them. But that's not right, we should be consistent and add documentation for every prop.

File: src/Badge/README.md [hunk 19]
```diff
@@ -10,22 +12,45 @@ notes: |
...
+$badge-font-weight:                 $font-weight-bold !default;
+$badge-padding-y:                   .125rem !default;
+$badge-padding-x:                   .5rem !default;
+$badge-border-radius:               .25rem !default;
+
+$badge-transition:                  none !default;
+$badge-focus-width:                 $input-btn-focus-width !default;
+
+$badge-pill-padding-x:              .6em !default;
+$badge-pill-border-radius:          10rem !default;
```
Comment (adamstankiewicz) [inline src/Badge/README.md:55]: The theme variables are no longer necessary in the README with the [changes recently merged](https://github.com/edx/paragon/pull/931). Just a symptom of a delay in reviewing this PR, my apologies.

File: src/Breadcrumb/README.md [hunk 20]
```diff
@@ -42,10 +44,29 @@ notes: |
...
+    { label: 'Link 2', url: '#link-2' },
+    { label: 'Link 3', url: '#link-3' },
   ]}
   spacer={<span className="custom-spacer">/</span>}
 />
 ```
+
+### Theme variables (SCSS)
+
+```scss
```
Comment (adamstankiewicz) [inline src/Breadcrumb/README.md:57]: The theme variables are no longer necessary in the README with the [changes recently merged](https://github.com/edx/paragon/pull/931). Just a symptom of a delay in reviewing this PR, my apologies.

File: src/Button/button-group.mdx (hunk 1 above)
Comment (adamstankiewicz) [reply src/Button/button-group.mdx:3]: I agree, one page should be enough here.

File: src/Carousel/README.md [hunk 21]
```diff
@@ -58,3 +62,31 @@ notes: |
   </Carousel.Item>
 </Carousel>
 ```
+
+### Theme Variables (SCSS)
```
Comment (adamstankiewicz) [inline src/Carousel/README.md:66]: The theme variables are no longer necessary in the README with the [changes recently merged](https://github.com/edx/paragon/pull/931). Just a symptom of a delay in reviewing this PR, my apologies.

File: src/Collapsible/README.md [hunk 22]
```diff
@@ -223,4 +229,18 @@ an **uncontrolled** Collapsible by getting a ref to the component and calling
 the component will be updated accordingly.
 
 
-`
+### Theme Variables (SCSS)
```
Comment (adamstankiewicz) [inline src/Collapsible/README.md:232]: The theme variables are no longer necessary in the README with the [changes recently merged](https://github.com/edx/paragon/pull/931). Just a symptom of a delay in reviewing this PR, my apologies.

**Review by adamstankiewicz — APPROVED** [review]
LGTM. We'll just need to remove the SCSS theme variables from the READMEs because of the other PR that recently shipped (🙃 )

**Comment by viktorrusakov** [comment]
@adamstankiewicz I removed theme variables from READMEs, thanks!

**Comment by edx-webhook** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/paragon&entry.867055334=https://github.com/edx/paragon/pull/898&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2021-11-22+17:10&entry.1917517419=2021-12-16+13:46&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**